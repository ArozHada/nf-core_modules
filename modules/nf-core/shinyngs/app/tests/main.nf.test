nextflow_process {

    name "Test Process SHINYNGS_APP"
    script "../main.nf"
    process "SHINYNGS_APP"

    tag "modules"
    tag "modules_nfcore"
    tag "shinyngs"
    tag "shinyngs/app"

    test("multi - tsv") {
        
        config "./multi.config"
        
        when {
            process {
                """
                expression_sample_sheet = file(params.test_data['mus_musculus']['genome']['rnaseq_samplesheet'], checkIfExists: true)             
                expression_feature_meta = file(params.test_data['mus_musculus']['genome']['rnaseq_genemeta'], checkIfExists: true)
                expression_contrasts = file(params.test_data['mus_musculus']['genome']['rnaseq_contrasts'], checkIfExists: true)

                // Load the same matrix twice to immitate multiple processings (e.g. raw
                // and normalised)                                                                                                                      
                                                                                                                                                         
                raw_expression_matrix_file = file(params.test_data['mus_musculus']['genome']['rnaseq_matrix'], checkIfExists: true)                     
                raw_expression_matrix_file.copyTo('normalised.tsv')                                                                                     
                normalised_expression_matrix_file = file('normalised.tsv')                                                                              
                                                                                                                                                         
                // Same for differential stats - we need two tables for the (now) two contrasts                                                         
                                                                                                                                                         
                expression_differential = file(params.test_data['mus_musculus']['genome']['deseq_results'], checkIfExists: true)                        
                expression_differential.copyTo('second_contrast_stats.tsv')                                                                             
                second_contrast_stats = file('second_contrast_stats.tsv')                                                                               
                
                input[0] = [
                    [ id:'test' ], // meta map
                    expression_sample_sheet,
                    expression_feature_meta,
                    [ raw_expression_matrix_file, normalised_expression_matrix_file ] 
                ]
                input[1] = [
                    [ "id":"SRP254919" ],
                    expression_contrasts,
                    [ expression_differential, second_contrast_stats  ]
                ]
                input[2] = 2
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
    test("single - tsv") {
        
        config "./single.config"

        when {
            process {
                """
                expression_sample_sheet = file(params.test_data['mus_musculus']['genome']['rnaseq_samplesheet'], checkIfExists: true) 
                expression_feature_meta = file(params.test_data['mus_musculus']['genome']['rnaseq_genemeta'], checkIfExists: true) 
                expression_matrix_file = file(params.test_data['mus_musculus']['genome']['rnaseq_matrix'], checkIfExists: true) 
                expression_contrasts = file(params.test_data['mus_musculus']['genome']['rnaseq_contrasts'], checkIfExists: true) 
                expression_differential = file(params.test_data['mus_musculus']['genome']['deseq_results'], checkIfExists: true) 

                // We need two tables for the (now) two contrasts 

                expression_differential = file(params.test_data['mus_musculus']['genome']['deseq_results'], checkIfExists: true) 
                expression_differential.copyTo('second_contrast_stats.tsv') 
                second_contrast_stats = file('second_contrast_stats.tsv')

                input[0] = [
                    [ id:'test' ], // meta map
                    expression_sample_sheet,
                    expression_feature_meta,
                    [ expression_matrix_file ] 
                ]
                input[1] = [
                    [ "id":"SRP254919" ],
                    expression_contrasts,
                    [ expression_differential, second_contrast_stats  ]
                ]
                input[2] = 1
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
