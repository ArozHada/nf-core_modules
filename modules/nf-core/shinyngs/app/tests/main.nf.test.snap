{
    "multi - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "data.rds:md5,105c36ec6ee81dd5f6a6f2b122b70208",
                        "app.R:md5,bedcfc45b6cdcc2b8fe3627987e2b17a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a8175cc675d6b84174fd1b394a5a4cfb"
                ],
                "app": [
                    [
                        {
                            "id": "test"
                        },
                        "data.rds:md5,105c36ec6ee81dd5f6a6f2b122b70208",
                        "app.R:md5,bedcfc45b6cdcc2b8fe3627987e2b17a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a8175cc675d6b84174fd1b394a5a4cfb"
                ]
            }
        ],
        "timestamp": "2024-01-07T21:24:15.138277"
    },
    "single - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "data.rds:md5,3c29bac9c8c3d13b740d47cfd0de0434",
                        "app.R:md5,bedcfc45b6cdcc2b8fe3627987e2b17a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a8175cc675d6b84174fd1b394a5a4cfb"
                ],
                "app": [
                    [
                        {
                            "id": "test"
                        },
                        "data.rds:md5,3c29bac9c8c3d13b740d47cfd0de0434",
                        "app.R:md5,bedcfc45b6cdcc2b8fe3627987e2b17a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a8175cc675d6b84174fd1b394a5a4cfb"
                ]
            }
        ],
        "timestamp": "2024-01-07T21:26:49.827522"
    }
}